CMakeLists.txt

```diff
@@ -93,6 +94,12 @@ endmacro (install_hook)
 # all setup common to the OPM library modules is done here
 include (OpmLibMain)
 
+# LIBECL is renamed to resdata (https://github.com/equinor/resdata)
+# To build opmflowdiagnosticsapplication you will need to install an old version of libecl 
+#include_directories(${CMAKE_SOURCE_DIR}/../ecl/lib/include)
+#include_directories(${CMAKE_SOURCE_DIR}/../ecl/build/lib/include)
+#target_link_libraries(opmflowdiagnosticsapplications PUBLIC ecl)
+
```

Comment:
Ideally we should rewrite the code to use the i/o from opm-common. Or second best to rename to use resdata. Currently you will need to have an old version of libecl on you system.

Reply:
Renaming to resdata shouldn't be hard I think. https://github.com/equinor/resdata/commit/1408dfb97f94cb8096a78ba9747ecf4c6e1dbe75

@bska How much work would it be to start using the opm-common i/o code? Does opm-common have the necessary features implemented?

Reply:
In `ResInsight,` to reduce the data imported from disk, we would like to import and work with a subset of cells. Typically this will be the IJK box covering all active cells. It would be useful to compute flow diagnostics data in this scenario also.

Reply:
> How much work would it be to start using the opm-common i/o code?

Sadly, "a lot". 

> Does opm-common have the necessary features implemented?

Not really, no.  The low-level readers will load the requisite linearised arrays from various sources, but LibECL/resdata additionally computes neighbourship information&ndash;including within local grids (LGRs) and between LGRs and the main grid.  We would have to build that connectivity graph ourselves here before we could switch to using the OPM-Common readers.

> It would be useful to compute flow diagnostics data [for subgrids] also.

That's an even bigger job, since it's not at all clear what that even means.  Do we impose no-flow conditions on the boundary of the subgrid?  If so, how do we adjust sources&i.e., flow rates at the well connection level&ndash;to account for fluid which no longer enters/leaves the subgrid? If not, what flow rates communicate between the subgrid and the rest of the model?

Reply:
> Sadly, "a lot".

Then I think I will do a renaming to support resdata. Depending on old libecl is a bit problematic for new users.